File: Niveau 3/3 – Gestion de caractères/6 - Validité des noms de variables.py
import sys
def main():
    charAccent = ['Ç', 'Ä', 'Å', 'É', 'Ö', 'Ü', 'Ñ', 'À', 'È', 'å', 'ì', 'ò', 'ÿ', 'á', 'í', 'ó', 'ú', 'ñ', 'é', 'à', 'è']
    nbNoms = int(input())
    for i in range(nbNoms):
        nom = input()
        valid = True
        if nom[0].isdigit():
            valid = False
        for c in nom:
            if not(c.isdigit() or c.isalpha() or c == '_') or c in charAccent:
                valid = False
        if valid:
            print("YES")
        else:
            print("NO")

main()

File: Niveau 3/03 – Gestion de caractères/6 - Validité des noms de variables.cpp
#include <iostream>
#include <cctype>
using namespace std;

bool isValidVariableName(string s)
{
  if (!isalpha(s[0]) && s[0] != '_')
  {
    return false;
  }
  for (int i = 1; i < (int)s.length(); i++)
  {
    if (!isalpha(s[i]) && s[i] != '_' && !isdigit(s[i]))
    {
      return false;
    }
  }
  return true;
}

int main()
{
  ios::sync_with_stdio(false);
  int n;
  cin >> n;
  for (int i = 0; i < n; i++)
  {
    string nom;
    cin >> nom;
    if (isValidVariableName(nom))
    {
      cout << "YES"
           << "\n";
    }
    else
    {
      cout << "NO"
           << "\n";
    }
  }
}
